src/main/java/org/testng/DependencyMap.java

```diff
@@ -61,12 +61,15 @@ public ITestNGMethod getMethodDependingOn(String methodName, ITestNGMethod fromM
     }
     for (ITestNGMethod m : l) {
       // If they are in the same class hierarchy, they must belong to the same instance,
+      Object fromMethodInstance = fromMethod.getInstance();
+      Object mInstance = m.getInstance();
+      if (mInstance != null && fromMethodInstance != null &&
+              fromMethodInstance.getClass().isAssignableFrom(mInstance.getClass())) {
+          return m;
+      } else {
       // otherwise, it's a method depending on a method in a different class so we
       // don't bother checking the instance
-      if (fromMethod.getRealClass().isAssignableFrom(m.getRealClass())) {
-        if (m.getInstance() == fromMethod.getInstance()) return m;
-      } else {
-        return m;
+          return m;
```

Comment:
indentation is only 2 spaces in this file.

Reply:
Yes I know. But its hard to follow that indentation especially in other places the IDE is configured to 4 spaces. So I chose to leave this as is. I think on a long term basis, we need to get this sorted out across all classes which brings us back to our earlier discussion on introducing a IDE formatter file for both IntelliJ and eclipse.